From c8e3f559f6217956a5e90b40b3a53406b908536a Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 2 Feb 2018 12:05:47 +0000
Subject: [PATCH 3/3] lib: Add private snapd directory to AoT path

snapd doesn't properly work with socket activation, and will only load
the profiles when snapd itself is started. Starting any snap before snapd
started will not have the profile labels available, thus without this uh,
solution, snaps won't work.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/lib/hook.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/lib/hook.c b/src/lib/hook.c
index c6afd6d..7642f36 100644
--- a/src/lib/hook.c
+++ b/src/lib/hook.c
@@ -40,9 +40,8 @@ static const char *apparmor_binaries[] = {
  * In future we intend to make this support stateless paths, pending work
  * upstream.
  */
-static const char *apparmor_directories[] = {
-        "/etc/apparmor.d",
-};
+static const char *apparmor_directories[] = { "/etc/apparmor.d",
+                                              "/var/lib/snapd/apparmor/profiles" };
 
 bool aa_hook_context_init(AaHookContext *self)
 {
@@ -100,4 +99,4 @@ bool aa_hook_context_init(AaHookContext *self)
  *
  * vi: set shiftwidth=8 tabstop=8 expandtab:
  * :indentSize=8:tabSize=8:noTabs=true:
- */
\ No newline at end of file
+ */
-- 
2.16.1

